volatility3/framework/plugins/windows/info.py

```diff
@@ -87,6 +103,78 @@ def get_kdbg_structure(cls, context: interfaces.context.ContextInterface, config
 
         return kdbg
 
+    @classmethod
+    def is_kdbg_encoded(cls, context: interfaces.context.ContextInterface, layer_name: str, symbol_table: str) -> bool:
+        ntkrnlmp = cls.get_kernel_module(context, layer_name, symbol_table)
+        if not ntkrnlmp.has_symbol("KdpDataBlockEncoded"):
+            return False
+
+        KdpDataBlockEncoded_value = ntkrnlmp.object("char", offset=ntkrnlmp.get_symbol("KdpDataBlockEncoded").address)
+
+        return KdpDataBlockEncoded_value != 0
+
+    @classmethod
+    def _decode_encoded_bytes(cls, context: interfaces.context.ContextInterface , offset: int,
+                              size: int, layer_name: str, symbol_table: str) -> bytes:
+
+        kernel = cls.get_kernel_module(context, layer_name, symbol_table)
+        wait_never = kernel.object("unsigned long long", offset=kernel.get_symbol("KiWaitNever").address)
+        wait_always = kernel.object("unsigned long long", offset=kernel.get_symbol("KiWaitAlways").address)
+        datablockencoded = kernel.object("char", offset=kernel.get_symbol("KdpDataBlockEncoded").address)
+
+        decoded_buffer = b""
+        encoded_array = kernel.object(object_type="array", subtype=kernel.get_type("unsigned long long"), offset=offset, layer_name=layer_name, count=(size // 8), absolute=True)
+        for entry in encoded_array:
+            low_byte = (wait_never) & 0xFF
+            entry = rol(entry ^ wait_never, low_byte)
+            # TODO: remove the OR after #702 gets merged in.
+            swap_xor = datablockencoded.vol.offset | 0xFFFF000000000000
+            entry = bswap(entry ^ swap_xor)
+            decoded_buffer += struct.pack("Q", entry ^ wait_always)
+
+        return decoded_buffer
+
+    @classmethod
+    def _create_structure_from_bytes(cls, context: interfaces.context.ContextInterface, 
+                                     structure_bytes: bytes, structure_symbol_table: str, structure_name: str, 
+                                     offset: int, native_layer_name: str):
+
+        new_layer_name = f"{offset}_{structure_name}"
```

Comment:
I can't tell whether it's better to have the name start with a number of a string.  I feel drawn towards string first, since the offset is a more specific differentiator.  Otherwise this seems like quite a neat way of handling data modification.

Reply:
replaced the order